src/remote_common.c

```diff
@@ -0,0 +1,336 @@
+    if (!die_on_amqp_error(amqp_get_rpc_reply(*conn), "Declaring the queue"))
+        return false;
+
+    return true;
+}
+
+bool declare_callback_queue(amqp_connection_state_t *conn,
+                            amqp_channel_t channel,
+                            amqp_bytes_t *reply_to_queue)
+{
+    /* Declare a exclusive private queues with TTL = 10s */
+    amqp_table_entry_t entries[1];
+    amqp_table_t table;
+    entries[0].key = amqp_cstring_bytes("x-message-ttl");
+    entries[0].value.kind = AMQP_FIELD_KIND_U32;
+    entries[0].value.value.u32 = 10 * 1000;  // 10s
+    table.num_entries = 1;
+    table.entries = entries;
+
+    amqp_queue_declare_ok_t *r =
+        amqp_queue_declare(*conn, channel, amqp_empty_bytes, 0, 0, 1, 0, table);
+    if (!die_on_amqp_error(amqp_get_rpc_reply(*conn),
+                           "Declaring the private queue with TTL = 10s"))
+        return false;
+
+    *reply_to_queue = amqp_bytes_malloc_dup(r->queue);
+    return true;
+}
+
+bool set_consuming_queue(amqp_connection_state_t *conn,
+                         amqp_channel_t channel,
+                         char const *queue_name)
+{
+    amqp_basic_consume(*conn, channel, amqp_cstring_bytes(queue_name),
+                       amqp_empty_bytes, 0, 0, 0, amqp_empty_table);
+    if (!die_on_amqp_error(amqp_get_rpc_reply(*conn),
+                           "Set the consuming queue"))
+        return false;
+
+    return true;
+}
+
+bool consume_message(amqp_connection_state_t *conn,
+                     amqp_channel_t channel,
+                     amqp_envelope_t *envelope)
+{
+    amqp_maybe_release_buffers(*conn);
+    /* Consume a message from the queue in the rabbitmq broker */
+    if (!die_on_amqp_error(amqp_consume_message(*conn, envelope, NULL, 0),
+                           "Consuming the message"))
+        return false;
+
+    return true;
+}
+
+bool wait_response_message(amqp_connection_state_t *conn,
+                           amqp_channel_t channel,
+                           amqp_bytes_t callback_queue,
+                           char *frame_body,
+                           int body_len)
+{
+    amqp_frame_t frame;
+#if defined(ENABLE_DEBUG)
+    amqp_basic_deliver_t *d;
+    amqp_basic_properties_t *p;
+#endif
+    size_t body_target;
+    size_t body_received;
+
+    amqp_basic_consume(*conn, channel, callback_queue, amqp_empty_bytes, 0, 1,
+                       0, amqp_empty_table);
+    if (!die_on_amqp_error(amqp_get_rpc_reply(*conn), "Set the callback queue"))
+        return false;
+
+    /*Wait for three frames: AMQP_FRAME_METHOD, AMQP_FRAME_HEADER,
+     * AMQP_FRAME_BODY*/
+    for (;;) {
+        amqp_maybe_release_buffers(*conn);
+        amqp_simple_wait_frame(*conn, &frame);
+        if (!die_on_amqp_error(amqp_get_rpc_reply(*conn), "Wait method frame"))
+            return false;
+
+#if defined(ENABLE_DEBUG)
+        printf(MSG_PREFIX "Frame type: %u channel: %u\n", frame.frame_type, frame.channel);
```

Comment:
We might introduce new inline function/macro `dprint` which follows the presence of `ENABLE_DEBUG` to wrap `printf`. @marktwtn, any idea?

Reply:
How about adding these macros in `src/common.h`:
```
#if defined(ENABLE_DEBUG)
    #define DPRINTF(...) printf(__VA_ARGS__)
#else
    #define DPRINTF(...)
#endif
```